Payments task: include Mercado Pago

octaedro opened the pull request:
Fixes #6537

This PR adds Mercado Pago as a payment method.

### Accessibility

<!-- If you've changed or added any interactions, check off the appropriate items below. You can delete any that don't apply. Use this space to elaborate on anything if needed. -->

-   [x] I've tested using only a keyboard (no mouse)
-   [ ] I've tested using a screen reader
-   [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
-   [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots
![screenshot-one wordpress test-2021 03 12-10_57_36](https://user-images.githubusercontent.com/1314156/110949852-dd023180-8321-11eb-9ec3-8f368a35db97.png)

![screenshot-one wordpress test-2021 03 12-10_13_01](https://user-images.githubusercontent.com/1314156/110945274-3cf5d980-831c-11eb-9af8-77bc0c58bebc.png)


### Detailed test instructions:
_Be sure to be using the latest WC (`trunk` branch) version in order to have the `Mercado Pago` logo available._

- Create a brand new store.
- Set one of the following countries in the first OBW step:
```
Mexico
Brazil
Argentina
Chile
Colombia
Peru
Uruguay
```
- Continue with the OBW and finish it up. 
- Select `Choose payment methods` in the setup task list (`Get ready to start selling`).
- Press the `Setup` button in the `Mercado Pago Payments` box.
- Try the links presented after the plugin's installation and verify they are working.
- Confirm that the `Mercado Pago payments for WooCommerce` plugin was installed.
- Press `Continue`.
- Now the `Mercado Pago Payments` option should appear as active.


<!-- 
Please add your test instructions to `/TESTING-INSTRUCTIONS.md`.
-->

<!--- Please add a Changelog note

Enter a changelog note following the WooCommerce core format using prefixes of Enhancement:, Tweak:, Dev:, Fix:, Performance: to readme.txt under the "unreleased" list at the top of the changelog. If none exists, please add it. Also include PR number.

If changes pertain to a package, also update CHANGELOG.md in the package's folder in a similar manner.--->

octaedro committed:
Added tests

octaedro committed:
Added plugin's name to constants

octaedro committed:
Added default url

octaedro committed:
Modified card copy

octaedro committed:
Modified copy

octaedro committed:
Added changelog

# Conflicts:
#	readme.txt

octaedro committed:
Added testing instructions

# Conflicts:
#	TESTING-INSTRUCTIONS.md

samueljseay commented:
@octaedro Are you aware the changelog label is just meant as a reminder? It will be self-removed when you add one. Also I notice this has "needs review" but also "in progress" is it ready for a review?

adrianduffell commented on client/task-list/tasks/payments/methods.js:
```diff
@@ -95,6 +96,32 @@ export function getPaymentMethods( {
+				</>
+			),
+			before: (
+				<img src={ wcAssetUrl + 'images/mercadopago.png' } alt="" />
```
This image didn't exist for me. Does it need to be committed in the PR?

octaedro replied:
Hi @adrianduffell,

> This image didn't exist for me.

Do you have the latest `trunk` version in the `WooCommerce` plugin you're using to test?
The image was merged a few days ago in [this PR](https://github.com/woocommerce/woocommerce/pull/29365).

> Does it need to be committed in the PR?

I added the image there to follow the pattern we are using for the `.png` logos in the payments methods (all of them are in the WooCommerce folder `assets/images/`).

Do you think we should start to store these images in a local folder?

adrianduffell replied:
@octaedro Ah, I had core 5.1 installed instead of the latest `trunk`. I guess this exposes an issue with the pattern where the image might not be available to WC Admin due to the version of core in use.

I think it would work well to have these images in a local WC Admin folder. Maybe there is a good reason to have them in core though?

octaedro replied:
> I think it would work well to have these images in a local WC Admin folder. Maybe there is a good reason to have them in core though?

Good idea! I couldn't think of any good reason to keep these images in `core`, so I put this PNG under `onboarding` in the commit 4044c5d.

adrianduffell reviewed (dismissed):
@octaedro  Nice work on this! Overall it looks good and tested well for me.

I just have one question about the logo. It didn't show up for me in the Payment Methods list.

octaedro committed:
Added "Local Partner" ribbon

octaedro commented:
As it was [requested today](https://github.com/woocommerce/woocommerce-admin/issues/6537#issuecomment-800351691), I also added a ribbon (similar to what we have [for Stripe](https://user-images.githubusercontent.com/1314156/111324770-d63d2c80-8649-11eb-99f0-f64c51dee87a.png)) but with the copy `Local Partner` in the commit 93c5fdd.

octaedro committed:
Using local image

octaedro commented:
Hi @adrianduffell,

I moved the image as you suggested and modified the reference. Could you give this PR another look?

becdetat commented on client/task-list/tasks/payments/mercadopago.js:
```diff
@@ -0,0 +1,100 @@
+};
+
+const MercadoPagoCredentialsStep = ( { countryCode, onFinish } ) => {
+	const getregistrationURL = () => {
```
should be `getRegistrationURL`

becdetat reviewed (approved):
Looks good, I  just found a typo, but other than that (and the conflicts) everything looks good. Pre-approving this.

octaedro committed:
Fixed typo

octaedro committed:
Added Mercado Pago payment gateway

# Conflicts:
#	client/task-list/tasks/payments/index.js
#	client/task-list/tasks/payments/methods.js

octaedro committed:
Fixed line duplication
